packages/jest-cli/src/__tests__/notify_reporter.test.js

```diff
@@ -0,0 +1,116 @@
+/**
+ * Copyright (c) 2014-present, Facebook, Inc. All rights reserved.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ */
+
+'use strict';
+
+import TestScheduler from '../test_scheduler';
+import NotifyReporter from '../reporters/notify_reporter';
+import type {TestSchedulerContext} from '../test_scheduler';
+import type {AggregatedResult} from '../../../../types/TestResult';
+
+jest.mock('../reporters/default_reporter');
+jest.mock('node-notifier', () => ({
+  notify: jest.fn(),
+}));
+
+const initialContext: TestSchedulerContext = {
+  firstRun: true,
+  previousSuccess: false,
+};
+
+const aggregatedResultsSuccess: AggregatedResult = {
+  numFailedTestSuites: 0,
+  numFailedTests: 0,
+  numPassedTestSuites: 1,
+  numPassedTests: 3,
+  numRuntimeErrorTestSuites: 0,
+  numTotalTestSuites: 1,
+  numTotalTests: 3,
+  success: true,
+};
+
+const aggregatedResultsFailure: AggregatedResult = {
+  numFailedTestSuites: 1,
+  numFailedTests: 3,
+  numPassedTestSuites: 0,
+  numPassedTests: 9,
+  numRuntimeErrorTestSuites: 0,
+  numTotalTestSuites: 1,
+  numTotalTests: 3,
+  success: false,
+};
+
+// Simulated sequence of events for NotifyReporter
+const notifyEvents = [
+  aggregatedResultsSuccess,
+  aggregatedResultsFailure,
+  aggregatedResultsSuccess,
+  aggregatedResultsSuccess,
+  aggregatedResultsFailure,
+  aggregatedResultsFailure,
+];
+
+test('.addReporter() .removeReporter()', () => {
+  const scheduler = new TestScheduler(
+    {},
+    {},
+    Object.assign({}, initialContext),
+  );
+  const reporter = new NotifyReporter();
+  scheduler.addReporter(reporter);
+  expect(scheduler._dispatcher._reporters).toContain(reporter);
+  scheduler.removeReporter(NotifyReporter);
+  expect(scheduler._dispatcher._reporters).not.toContain(reporter);
+});
+
+const testModes = (notifyMode: string, arl: Array<AggregatedResult>) => {
+  const notify = require('node-notifier');
+  notify.notify.mock.calls = [];
```

Comment:
mock.calls contains the calls of my previous tests. Is there a better way to do this?

Reply:
Call [`jest.clearAllMocks()`](https://facebook.github.io/jest/docs/en/jest-object.html#jestclearallmocks) in `afterEach`